### PR TITLE
MSVC fixes

### PR DESCRIPTION
**Pull request by vasama**
This is a set of fixes targeting intended behaviours of MSVC. I'm holding off on any compiler bug workarounds for now, as fixes for some of the bugs are pending release or the bugs may be fixed before I can find and work around the rest.

### CODE REVIEW

File: include/exec/__detail/__sender_facade.hpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ namespace exec {
       if constexpr (__lacks_transform_sender<_Kernel>) {
         return __mtype<_Sender>{};
       } else {
-        if constexpr (__valid<__tfx_sender_, _Kernel, _Sender, _Env>) {
+        if constexpr (__valid_<__tfx_sender_, _Kernel, _Sender, _Env>) {
```
Comment (ericniebler) [inline include/exec/__detail/__sender_facade.hpp:112]: what does `__valid` mean on msvc?

**Review by ericniebler — APPROVED** [review]

File: include/exec/__detail/__sender_facade.hpp (hunk 1 above)
Comment (ericniebler) [reply include/exec/__detail/__sender_facade.hpp:112]: Ah. It's in `<sal.h>` as well: https://gist.github.com/caiorss/0678b5efb2aa2eff120b9e9284fdbb0c#file-sal-h-L2448-L2460

Comment (ericniebler) [reply include/exec/__detail/__sender_facade.hpp:112]: Could you rename `__valid` instead to `__mvalid`?

Comment (vasama) [reply include/exec/__detail/__sender_facade.hpp]: Done.

**Comment by ericniebler** [comment]
Thanks!

**Pull request merged**